websites/N/Neko-sama.fr/presence.ts

```diff
@@ -0,0 +1,104 @@
+const presence = new Presence({
+	clientId: "1014298173314961481",
+});
+
+function timeToString(nbr: number): string {
+	let nbrCopy = nbr,
+		nbrString = "",
+		quotient = 0,
+		remainder = 0;
+	if (nbrCopy >= 3600) {
+		quotient = Math.floor(nbrCopy / 3600);
+		if (isNaN(quotient)) quotient = 0;
+		remainder = nbrCopy % 3600;
+		if (quotient > 9) nbrString += `${quotient.toString()}:`;
+		else nbrString += `0${quotient.toString()}:`;
+
+		nbrCopy = remainder;
+	}
+	quotient = Math.floor(nbrCopy / 60);
+	if (isNaN(quotient)) quotient = 0;
+	remainder = nbrCopy % 60;
+	if (quotient > 9) nbrString += `${quotient.toString()}:`;
+	else nbrString += `0${quotient.toString()}:`;
+
+	nbrCopy = remainder;
+	if (isNaN(nbrCopy)) nbrCopy = 0;
+	if (nbrCopy > 9) nbrString += nbrCopy.toString();
+	else nbrString += `0${nbrCopy.toString()}`;
+
+	return nbrString;
+}
+
+interface Video {
+	time: number;
+	duration: number;
+	paused: boolean;
+}
+
+let video: Video = null;
+
+presence.on("iFrameData", (data: Video) => {
+	video = data;
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "nekosama-icon",
+	};
+
+	if (document.URL.startsWith("https://www.neko-sama.fr/anime/episode/")) {
+		const episodeImage: string =
+				document.querySelector<HTMLMetaElement>('meta[property="og:image"]')
+					.content === "https://neko-sama.fr/images/default_thumbnail.png"
+					? "nekosama-icon"
+					: document.querySelector<HTMLMetaElement>('meta[property="og:image"]')
+							.content,
+			{ paused, time, duration } = video;
+		if (!paused) {
+			const timestamps = presence.getTimestamps(time, duration);
+			presenceData.startTimestamp = timestamps[0];
+			presenceData.endTimestamp = timestamps[1];
+		}
+		presenceData.state = `${timeToString(Math.floor(time))}/${timeToString(
+			Math.floor(duration)
+		)}`;
```

Comment:
What's the point of showing the time as a string in the state?

Reply:
I made that because for me just getting lefting time isn't enough, we don't know how long episode last and where user is on it.
I can remove timestamps because I prefer MM:SS/MM:SS format but I don't know which user prefers.

Reply:
@EncryptedDev ^